### PR TITLE
Compare work

### PR DESCRIPTION
**Pull request by bhollis**
Did some work on Compare:

1. Did some profiling after noticing that highlighting stats on mouseover felt laggy. 10x rendering speedup from ~30ms to 3ms on my machine.
2. Fix the persistent vertical scrollability that causes things to misalign. New rule: always specify both `overflow-x` and `overflow-y` if you specify one.
3. Remove a weird restriction on the perk preview function where it'd only allow sockets with index < 2 to be clicked.
4. Fix the sizing of perks in D1, they were cut off.
5. Add a new `locateItem` function and custom implementation for both Desktop and Phone, with the phone one switching to the correct character before popping the item.

### CODE REVIEW

File: src/app/inventory/PhoneStores.tsx [hunk 1]
```diff
@@ -35,6 +37,29 @@ export default function PhoneStores({ stores, buckets, singleCharacter }: Props)
   const [selectedCategoryId, setSelectedCategoryId] = useState<string>('Weapons');
   const detachedLoadoutMenu = useRef<HTMLDivElement>(null);
 
+  // Handle scrolling the right store into view when locating an item
+  useEventBusListener(
```
Comment (kyleshay) [inline src/app/inventory/PhoneStores.tsx:41]: Ooh, this is cool!

**Comment by kyleshay** [comment]
Interesting about having to specify both overflows, good to know

**Review by nev-r — APPROVED** [review]
seems snappy! not sure this actually fixes alignment, but maybe it's differently wrong on different clients
image

**Comment by bhollis** [comment]
The alignment itself is super jank, based on a hardcoded spacer height. The thing that's fixed is that it won't scroll vertically anymore.

**Pull request merged**